query-node/mappings/src/workingGroup.ts

```diff
@@ -134,27 +159,54 @@ export async function workingGroup_WorkerStorageUpdated(db: DatabaseManager, wor
   await db.save<Worker>(worker)
 
   // emit log event
-  logger.info("Storage provider has been updated", {workerId, workerType})
+  logger.info("Worker has been updated", {workerId, workerType})
 }
 
 export async function workingGroup_TerminatedWorker(db: DatabaseManager, workerType: WorkerType, workerId: WorkerId): Promise<void> {
   // do removal logic
   await deactivateWorker(db, workerType, workerId)
 
   // emit log event
-  logger.info("Storage provider has beed removed (worker terminated)", {workerId, workerType})
+  logger.info("Worker has been removed (worker terminated)", {workerId, workerType})
 }
 
 export async function workingGroup_WorkerExited(db: DatabaseManager, workerType: WorkerType, workerId: WorkerId): Promise<void> {
   // do removal logic
   await deactivateWorker(db, workerType, workerId)
 
   // emit log event
-  logger.info("Storage provider has beed removed (worker exited)", {workerId, workerType})
+  logger.info("Worker has been removed (worker exited)", {workerId, workerType})
+}
+
+export async function workingGroup_LeaderSet(db: DatabaseManager, workerType: WorkerType, workerId: WorkerId): Promise<void> {
```

Comment:
We should not create a worker when handling LeaderSet event. Because the OpeningFilled event will also be triggered later for the same worker. So there will be two Workers created for leads. I suggest we either drop handling of LeaderSet event all together, or in the OpeningFilled event we look for same workerId already existing and not create a new worker. With latter approach we can risk hiding incorrect handling of when workers are removed however.

Reply:
I overlooked that. LeaderSet event was removed.